[ska] relocate common services and utils to security_solution_api_integration dir

dmlemeshko opened the pull request:
## Summary

It is a preparation work before we relocate `x-pack/test/security_solution_api_integration` under `x-pack/solutions/security/test`:

- moving helpers/utils inside directory to avoid updating imports after relocation
- importing from `security_solution_api_integration` inside `x-pack/solutions/security/test` leads to circular dependency issue, so I copied `common/utils/detections_response` as a temporary workaround until tests dir is relocated

<!--ONMERGE {"backportTargets":["8.19","9.1"]} ONMERGE-->

dmlemeshko committed:
[ska] relocate common services and utils to security_solution_api_integration

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321388)
* Commit: b854aa31991311c339465af48b04e9d352bfbc56

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/321388#01982d0d-87b9-44f7-a666-c1a1ed636885)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321388","number":321388,"commit":"b854aa31991311c339465af48b04e9d352bfbc56"}],"number":321388}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321420)
* Commit: b648a4165875b351bba954a7d78079b169659882

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/321420#01982d41-6c41-4147-a2a6-3e9d7bfb04b2)

### History
* :broken_heart: [Build #321388](https://buildkite.com/elastic/kibana-pull-request/builds/321388) failed b854aa31991311c339465af48b04e9d352bfbc56

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321420","number":321420,"commit":"b648a4165875b351bba954a7d78079b169659882"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321388","number":321388,"commit":"b854aa31991311c339465af48b04e9d352bfbc56"}],"number":321420}
buildkite-pr-comment-->

dmlemeshko committed:
update imports

dmlemeshko committed:
fix imports in SecSol services

dmlemeshko committed:
fix more imports

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

dmlemeshko committed:
Update CODEOWNERS

dmlemeshko committed:
copy helpers to resolve circular deps issue

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321804)
* Commit: ccaaf7d6b6835c4ccff6a795205727b17b9d1a4b

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/321804#019831db-32b9-4ab9-afa6-068cc0f4b128)

### History
* :broken_heart: [Build #321762](https://buildkite.com/elastic/kibana-pull-request/builds/321762) failed 0908e6594b1ea3094d42b7555de117facfd83a10
* :broken_heart: [Build #321697](https://buildkite.com/elastic/kibana-pull-request/builds/321697) failed 8be234192750302fbf3ac92fe908d741f92205a5
* :broken_heart: [Build #321420](https://buildkite.com/elastic/kibana-pull-request/builds/321420) failed b648a4165875b351bba954a7d78079b169659882
* :broken_heart: [Build #321388](https://buildkite.com/elastic/kibana-pull-request/builds/321388) failed b854aa31991311c339465af48b04e9d352bfbc56

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321804","number":321804,"commit":"ccaaf7d6b6835c4ccff6a795205727b17b9d1a4b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321762","number":321762,"commit":"0908e6594b1ea3094d42b7555de117facfd83a10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321697","number":321697,"commit":"8be234192750302fbf3ac92fe908d741f92205a5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321420","number":321420,"commit":"b648a4165875b351bba954a7d78079b169659882"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321388","number":321388,"commit":"b854aa31991311c339465af48b04e9d352bfbc56"}],"number":321804}
buildkite-pr-comment-->

dmlemeshko committed:
fix imports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321915)
* Commit: 8ed511b978867b3adb096509262973315d694adb

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/321915#0198328d-36b0-4c3b-bbc9-51341da8d011)

### History
* :broken_heart: [Build #321851](https://buildkite.com/elastic/kibana-pull-request/builds/321851) failed b3123b8b7e68927e45ab3e2ca6895887bad05b7a
* :broken_heart: [Build #321804](https://buildkite.com/elastic/kibana-pull-request/builds/321804) failed ccaaf7d6b6835c4ccff6a795205727b17b9d1a4b
* :broken_heart: [Build #321762](https://buildkite.com/elastic/kibana-pull-request/builds/321762) failed 0908e6594b1ea3094d42b7555de117facfd83a10
* :broken_heart: [Build #321697](https://buildkite.com/elastic/kibana-pull-request/builds/321697) failed 8be234192750302fbf3ac92fe908d741f92205a5
* :broken_heart: [Build #321420](https://buildkite.com/elastic/kibana-pull-request/builds/321420) failed b648a4165875b351bba954a7d78079b169659882

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321915","number":321915,"commit":"8ed511b978867b3adb096509262973315d694adb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321851","number":321851,"commit":"b3123b8b7e68927e45ab3e2ca6895887bad05b7a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321804","number":321804,"commit":"ccaaf7d6b6835c4ccff6a795205727b17b9d1a4b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321762","number":321762,"commit":"0908e6594b1ea3094d42b7555de117facfd83a10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321697","number":321697,"commit":"8be234192750302fbf3ac92fe908d741f92205a5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321420","number":321420,"commit":"b648a4165875b351bba954a7d78079b169659882"}],"number":321915}
buildkite-pr-comment-->

dmlemeshko committed:
fix import

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321935)
* Commit: 585f9a1433a8fbee086b0e3111f5a16aecca26b6

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/321935#019832ad-118b-4023-ad20-1c53c52cdc1b)

### History
* :broken_heart: [Build #321915](https://buildkite.com/elastic/kibana-pull-request/builds/321915) failed 8ed511b978867b3adb096509262973315d694adb
* :broken_heart: [Build #321851](https://buildkite.com/elastic/kibana-pull-request/builds/321851) failed b3123b8b7e68927e45ab3e2ca6895887bad05b7a
* :broken_heart: [Build #321804](https://buildkite.com/elastic/kibana-pull-request/builds/321804) failed ccaaf7d6b6835c4ccff6a795205727b17b9d1a4b
* :broken_heart: [Build #321762](https://buildkite.com/elastic/kibana-pull-request/builds/321762) failed 0908e6594b1ea3094d42b7555de117facfd83a10
* :broken_heart: [Build #321697](https://buildkite.com/elastic/kibana-pull-request/builds/321697) failed 8be234192750302fbf3ac92fe908d741f92205a5

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321935","number":321935,"commit":"585f9a1433a8fbee086b0e3111f5a16aecca26b6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321915","number":321915,"commit":"8ed511b978867b3adb096509262973315d694adb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321851","number":321851,"commit":"b3123b8b7e68927e45ab3e2ca6895887bad05b7a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321804","number":321804,"commit":"ccaaf7d6b6835c4ccff6a795205727b17b9d1a4b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321762","number":321762,"commit":"0908e6594b1ea3094d42b7555de117facfd83a10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321697","number":321697,"commit":"8be234192750302fbf3ac92fe908d741f92205a5"}],"number":321935}
buildkite-pr-comment-->

dmlemeshko committed:
copy services to fix TS error

dmlemeshko committed:
fix imports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/322531)
* Commit: fb03e24a9c6d7f84c155c2e569946929d9efbb8a

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/322531#01983804-ca5f-403c-b2d6-74b0427fdc53)

### History
* :broken_heart: [Build #322322](https://buildkite.com/elastic/kibana-pull-request/builds/322322) failed 8fe4f54bb802495f0e926f781567439b3270453a
* :broken_heart: [Build #322237](https://buildkite.com/elastic/kibana-pull-request/builds/322237) failed 81059b0ce11eee9578ca073ea36642dcde19ccf5
* :broken_heart: [Build #321935](https://buildkite.com/elastic/kibana-pull-request/builds/321935) failed 585f9a1433a8fbee086b0e3111f5a16aecca26b6
* :broken_heart: [Build #321915](https://buildkite.com/elastic/kibana-pull-request/builds/321915) failed 8ed511b978867b3adb096509262973315d694adb
* :broken_heart: [Build #321851](https://buildkite.com/elastic/kibana-pull-request/builds/321851) failed b3123b8b7e68927e45ab3e2ca6895887bad05b7a

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322531","number":322531,"commit":"fb03e24a9c6d7f84c155c2e569946929d9efbb8a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322322","number":322322,"commit":"8fe4f54bb802495f0e926f781567439b3270453a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322237","number":322237,"commit":"81059b0ce11eee9578ca073ea36642dcde19ccf5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321935","number":321935,"commit":"585f9a1433a8fbee086b0e3111f5a16aecca26b6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321915","number":321915,"commit":"8ed511b978867b3adb096509262973315d694adb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321851","number":321851,"commit":"b3123b8b7e68927e45ab3e2ca6895887bad05b7a"}],"number":322531}
buildkite-pr-comment-->

maximpn committed:
fix tsconfigs

maximpn committed:
fix failed CI

maximpn committed:
get rid of the local routeWithNamespace dependency

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/326883)
* Commit: d4fbfe49f7fc522ac44c6e2e30453a3814a1585b

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/326883#01987ed7-7fd0-4316-84a8-ffb14640d7ea)

### History
* :broken_heart: [Build #326815](https://buildkite.com/elastic/kibana-pull-request/builds/326815) failed f317bafa776f2299fe793cf5cc39b9968bac9294
* :broken_heart: [Build #326759](https://buildkite.com/elastic/kibana-pull-request/builds/326759) failed 103d879e3ec79390febe48a70c7103958713e83f
* :broken_heart: [Build #326612](https://buildkite.com/elastic/kibana-pull-request/builds/326612) failed 98db045c6ad1eb4608a9157044ddfd7badbc7217
* :broken_heart: [Build #326582](https://buildkite.com/elastic/kibana-pull-request/builds/326582) failed fb43d3443fa493ef37ed6c36932fdd3bec11d179
* :broken_heart: [Build #326535](https://buildkite.com/elastic/kibana-pull-request/builds/326535) failed 314fd400a2fe441811e1c4e249a1029f264c2e9b

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326883","number":326883,"commit":"d4fbfe49f7fc522ac44c6e2e30453a3814a1585b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326815","number":326815,"commit":"f317bafa776f2299fe793cf5cc39b9968bac9294"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326759","number":326759,"commit":"103d879e3ec79390febe48a70c7103958713e83f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326612","number":326612,"commit":"98db045c6ad1eb4608a9157044ddfd7badbc7217"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326582","number":326582,"commit":"fb43d3443fa493ef37ed6c36932fdd3bec11d179"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326535","number":326535,"commit":"314fd400a2fe441811e1c4e249a1029f264c2e9b"}],"number":326883}
buildkite-pr-comment-->

maximpn reviewed (approved):
@dmlemeshko I've rebased the PR after merging our recent integration test related changes. There was a little issue with using spaces helper function in auto-generated clients `security_solution_api.gen.ts`, `security_solution_endpoint_exceptions_api.gen.ts`, `security_solution_exceptions_api.gen.ts`, `security_solution_lists_api.gen.ts` and `security_solution_osquery_api.gen.ts` so I've addressed it.

The rest of the diff LGTM.

dmlemeshko committed:
Merge branch 'main' into ska/relocate-sec-solution-ftr-helpers

dmlemeshko commented:
> @dmlemeshko I've rebased the PR after merging our recent integration test related changes. There was a little issue with using spaces helper function in auto-generated clients `security_solution_api.gen.ts`, `security_solution_endpoint_exceptions_api.gen.ts`, `security_solution_exceptions_api.gen.ts`, `security_solution_lists_api.gen.ts` and `security_solution_osquery_api.gen.ts` so I've addressed it.
> 
> The rest of the diff LGTM.

Thank you @maximpn I appreciate your help here. Hope we can merge it after https://github.com/elastic/kibana/pull/230198

dmlemeshko committed:
relocate spaces FTR service

dmlemeshko committed:
remove synthtrace common ftr service

dmlemeshko committed:
delete search_secure common ftr service, update codeowners

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327145)
* Commit: 14bbfb647be4afccbab830f3f198bc5e009f79d6

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/327145#01988032-eaaf-4b86-9c3a-d361f501faf0)

### History
* :broken_heart: [Build #326925](https://buildkite.com/elastic/kibana-pull-request/builds/326925) failed 1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7
* :broken_heart: [Build #326883](https://buildkite.com/elastic/kibana-pull-request/builds/326883) failed d4fbfe49f7fc522ac44c6e2e30453a3814a1585b
* :broken_heart: [Build #326815](https://buildkite.com/elastic/kibana-pull-request/builds/326815) failed f317bafa776f2299fe793cf5cc39b9968bac9294
* :broken_heart: [Build #326759](https://buildkite.com/elastic/kibana-pull-request/builds/326759) failed 103d879e3ec79390febe48a70c7103958713e83f
* :broken_heart: [Build #326612](https://buildkite.com/elastic/kibana-pull-request/builds/326612) failed 98db045c6ad1eb4608a9157044ddfd7badbc7217

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327145","number":327145,"commit":"14bbfb647be4afccbab830f3f198bc5e009f79d6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326925","number":326925,"commit":"1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326883","number":326883,"commit":"d4fbfe49f7fc522ac44c6e2e30453a3814a1585b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326815","number":326815,"commit":"f317bafa776f2299fe793cf5cc39b9968bac9294"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326759","number":326759,"commit":"103d879e3ec79390febe48a70c7103958713e83f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326612","number":326612,"commit":"98db045c6ad1eb4608a9157044ddfd7badbc7217"}],"number":327145}
buildkite-pr-comment-->

dmlemeshko committed:
use platform services for cloud_security_posture_functional

dmlemeshko committed:
Merge branch 'ska/relocate-sec-solution-ftr-helpers' of github.com:dmlemeshko/kibana into ska/relocate-sec-solution-ftr-helpers

dmlemeshko committed:
Merge remote-tracking branch 'upstream/main' into ska/relocate-sec-solution-ftr-helpers

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327223)
* Commit: bced66dc56b2a313a92fa78d1d3aec762b4584c9

### Failed CI Steps
* [FTR Configs #34](https://buildkite.com/elastic/kibana-pull-request/builds/327223#01988133-a8f7-4543-9c0e-c893186aa8e5)
* [FTR Configs #72](https://buildkite.com/elastic/kibana-pull-request/builds/327223#01988133-a91b-4196-8052-ff77d8086b3a)
* [FTR Configs #95](https://buildkite.com/elastic/kibana-pull-request/builds/327223#01988133-a934-413c-b5d4-69fc4b6350b8)
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/327223#01988133-a953-4d77-ad4c-17339f179e81)
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/327223#0198815c-6dad-4e14-8f59-85a5a226db91)
* [FTR Configs #119](https://buildkite.com/elastic/kibana-pull-request/builds/327223#01988133-a95f-48b3-8f90-4061703549c1)
* [FTR Configs #133](https://buildkite.com/elastic/kibana-pull-request/builds/327223#01988133-a971-4a92-93d2-3887c1d53508)

### History
* :broken_heart: [Build #327145](https://buildkite.com/elastic/kibana-pull-request/builds/327145) failed 14bbfb647be4afccbab830f3f198bc5e009f79d6
* :broken_heart: [Build #326925](https://buildkite.com/elastic/kibana-pull-request/builds/326925) failed 1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7
* :broken_heart: [Build #326883](https://buildkite.com/elastic/kibana-pull-request/builds/326883) failed d4fbfe49f7fc522ac44c6e2e30453a3814a1585b
* :broken_heart: [Build #326815](https://buildkite.com/elastic/kibana-pull-request/builds/326815) failed f317bafa776f2299fe793cf5cc39b9968bac9294
* :broken_heart: [Build #326759](https://buildkite.com/elastic/kibana-pull-request/builds/326759) failed 103d879e3ec79390febe48a70c7103958713e83f

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327223","number":327223,"commit":"bced66dc56b2a313a92fa78d1d3aec762b4584c9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327145","number":327145,"commit":"14bbfb647be4afccbab830f3f198bc5e009f79d6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326925","number":326925,"commit":"1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326883","number":326883,"commit":"d4fbfe49f7fc522ac44c6e2e30453a3814a1585b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326815","number":326815,"commit":"f317bafa776f2299fe793cf5cc39b9968bac9294"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326759","number":326759,"commit":"103d879e3ec79390febe48a70c7103958713e83f"}],"number":327223}
buildkite-pr-comment-->

dmlemeshko committed:
load secureSearch

dmlemeshko committed:
Merge branch 'main' into ska/relocate-sec-solution-ftr-helpers

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327264)
* Commit: 8755cd14b5339585df773a77391e1f62ea16aee1

### Failed CI Steps
* [FTR Configs #79](https://buildkite.com/elastic/kibana-pull-request/builds/327264#019881ae-a3a7-4643-8230-caa6596859bf)
* [FTR Configs #79](https://buildkite.com/elastic/kibana-pull-request/builds/327264#019881d4-986c-4ebf-8385-52528c3595e5)

### History
* :broken_heart: [Build #327223](https://buildkite.com/elastic/kibana-pull-request/builds/327223) failed bced66dc56b2a313a92fa78d1d3aec762b4584c9
* :broken_heart: [Build #327145](https://buildkite.com/elastic/kibana-pull-request/builds/327145) failed 14bbfb647be4afccbab830f3f198bc5e009f79d6
* :broken_heart: [Build #326925](https://buildkite.com/elastic/kibana-pull-request/builds/326925) failed 1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7
* :broken_heart: [Build #326883](https://buildkite.com/elastic/kibana-pull-request/builds/326883) failed d4fbfe49f7fc522ac44c6e2e30453a3814a1585b
* :broken_heart: [Build #326815](https://buildkite.com/elastic/kibana-pull-request/builds/326815) failed f317bafa776f2299fe793cf5cc39b9968bac9294

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327264","number":327264,"commit":"8755cd14b5339585df773a77391e1f62ea16aee1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327223","number":327223,"commit":"bced66dc56b2a313a92fa78d1d3aec762b4584c9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327145","number":327145,"commit":"14bbfb647be4afccbab830f3f198bc5e009f79d6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326925","number":326925,"commit":"1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326883","number":326883,"commit":"d4fbfe49f7fc522ac44c6e2e30453a3814a1585b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326815","number":326815,"commit":"f317bafa776f2299fe793cf5cc39b9968bac9294"}],"number":327264}
buildkite-pr-comment-->

dmlemeshko committed:
use platform services in cloud_security_posture_functional

dmlemeshko commented on .github/CODEOWNERS:
```diff
@@ -1510,7 +1510,6 @@ x-pack/solutions/observability/plugins/observability/server/lib/esql_extensions
 /x-pack/platform/plugins/shared/streams_app/public/components/data_management @elastic/obs-ux-logs-team
 
 # Infra Monitoring tests
-/x-pack/test/common/services/synthtrace.ts @elastic/obs-ux-infra_services-team
```
service was relocated, removing the copy

dmlemeshko commented on .github/CODEOWNERS:
```diff
@@ -2153,7 +2154,6 @@ x-pack/platform/plugins/private/cloud_integrations/cloud_full_story/server/confi
 /x-pack/platform/test/functional/page_objects/api_keys_page.ts @elastic/kibana-security
 /x-pack/platform/test/functional/page_objects/account_settings_page.ts @elastic/kibana-security
 /x-pack/platform/test/functional/apps/user_profiles @elastic/kibana-security
-/x-pack/test/common/services/spaces.ts @elastic/kibana-security
```
service was relocated, removing the copy

dmlemeshko commented on .github/CODEOWNERS:
```diff
@@ -2287,7 +2287,6 @@ x-pack/platform/plugins/private/cloud_integrations/cloud_full_story/server/confi
 # Enterprise Search
 # search
 /x-pack/platform/test/fixtures/es_archives/data/search_sessions @elastic/search-kibana
-/x-pack/test/common/services/search_secure.ts @elastic/search-kibana
```
service was relocated, removing the copy

dmlemeshko commented:
run docs-build

csr reviewed (approved):
LGTM 👍

dmlemeshko committed:
Merge branch 'main' into ska/relocate-sec-solution-ftr-helpers

opauloh reviewed (approved):
Cloud Security Posture changes LGTM

spong reviewed (approved):
Security GenAI changes LGTM! 👍  Thanks for helping with this refactor @dmlemeshko! 🙂

rylnd reviewed (approved):
Detection Engine changes LGTM

dmlemeshko committed:
Merge remote-tracking branch 'upstream/main' into ska/relocate-sec-solution-ftr-helpers

dmlemeshko committed:
Merge branch 'main' into ska/relocate-sec-solution-ftr-helpers

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327791)
* Commit: e256847fa810341dc5669509840a901314d62254

### Failed CI Steps
* [FTR Configs #17](https://buildkite.com/elastic/kibana-pull-request/builds/327791#0198889a-7e76-4c45-b210-9071bf3bcbb1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/327791#0198889a-7e76-4c45-b210-9071bf3bcbb1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/327791/jobs/0198889a-7e76-4c45-b210-9071bf3bcbb1/artifacts/019888cd-c080-42a3-97dc-e74b4cf5374a) FTR Configs #<span></span>17 / Interactive setup APIs - Enrollment flow should be able to enroll with valid authentication code

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/0be6db10cea038b9d91c81d1f0a8f069507e045d) | [after](https://github.com/elastic/kibana/commit/e256847fa810341dc5669509840a901314d62254) | diff |
| --- | --- | --- | --- |
| `spaces` | 236 | 237 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/0be6db10cea038b9d91c81d1f0a8f069507e045d) | [after](https://github.com/elastic/kibana/commit/e256847fa810341dc5669509840a901314d62254) | diff |
| --- | --- | --- | --- |
| `spaces` | 28.5KB | 28.6KB | +67.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/0be6db10cea038b9d91c81d1f0a8f069507e045d) | [after](https://github.com/elastic/kibana/commit/e256847fa810341dc5669509840a901314d62254) | diff |
| --- | --- | --- | --- |
| `spaces` | 256 | 259 | +3 |


</details>


### History
* :yellow_heart: [Build #327745](https://buildkite.com/elastic/kibana-pull-request/builds/327745) was flaky b8e48adeb32b455c1e1b351cc0875cc9a28ffd6f
* :yellow_heart: [Build #327584](https://buildkite.com/elastic/kibana-pull-request/builds/327584) was flaky 1f79d7caeb4bd266f7280d752ca54dea76fd3975
* :broken_heart: [Build #327264](https://buildkite.com/elastic/kibana-pull-request/builds/327264) failed 8755cd14b5339585df773a77391e1f62ea16aee1
* :broken_heart: [Build #327223](https://buildkite.com/elastic/kibana-pull-request/builds/327223) failed bced66dc56b2a313a92fa78d1d3aec762b4584c9
* :broken_heart: [Build #327145](https://buildkite.com/elastic/kibana-pull-request/builds/327145) failed 14bbfb647be4afccbab830f3f198bc5e009f79d6
* :broken_heart: [Build #326925](https://buildkite.com/elastic/kibana-pull-request/builds/326925) failed 1d2519fefc3118baff0b9524ff8ce3bae4dcc4b7

cc @dmlemeshko

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327791","number":327791,"commit":"e256847fa810341dc5669509840a901314d62254"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327745","number":327745,"commit":"b8e48adeb32b455c1e1b351cc0875cc9a28ffd6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327584","number":327584,"commit":"1f79d7caeb4bd266f7280d752ca54dea76fd3975"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327264","number":327264,"commit":"8755cd14b5339585df773a77391e1f62ea16aee1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327223","number":327223,"commit":"bced66dc56b2a313a92fa78d1d3aec762b4584c9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327145","number":327145,"commit":"14bbfb647be4afccbab830f3f198bc5e009f79d6"}],"number":327791}
buildkite-pr-comment-->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 228780` locally
cc: @dmlemeshko

dmlemeshko commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.1|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/231623">](https://github.com/elastic/kibana/pull/231623)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.1 -->
